Src/GoogleApis/Apis/[Media]/Download/MediaDownloader.cs

```diff
@@ -206,6 +208,16 @@ public IDownloadProgress Download(string url, Stream stream)
                     using (var response = await service.HttpClient.SendAsync(request, cancellationToken).
                         ConfigureAwait(false))
                     {
+                        if (!response.IsSuccessStatusCode)
+                        {
+                            var error = await service.DeserializeError(response).ConfigureAwait(false);
+                            throw new GoogleApiException(service.Name, error.ToString())

```

Comment:
re: 1 - it tied into the query param issue. I was hitting a method that served media and took a query parameter that triggered the encoding bug. That produced a 400 error but Download() didn't return an error or throw an exception. It just downloaded the error response to the stream as if everything was normal :(